src/physics/operators/odes.hpp

```diff
@@ -0,0 +1,263 @@
+#include <functional>
+#include <variant>
+
+#include "mfem.hpp"
+#include "numerics/expr_template_ops.hpp"
+#include "physics/utilities/boundary_condition_manager.hpp"
+#include "physics/utilities/equation_solver.hpp"
+
+class FirstOrderODE : public mfem::TimeDependentOperator {
+public:
+  using explicit_signature = void(const double, const mfem::Vector&, mfem::Vector&);
+  using implicit_signature = void(const double, const double, const mfem::Vector&, mfem::Vector&);
+
+  FirstOrderODE(int n) : mfem::TimeDependentOperator(n, 0.0){};
+  void Mult(const mfem::Vector& u, mfem::Vector& du_dt) const { explicit_func(t, u, du_dt); }
+  void ImplicitSolve(const double dt, const mfem::Vector& u, mfem::Vector& du_dt) { implicit_func(t, dt, u, du_dt); }
+
+  std::function<explicit_signature> explicit_func;
+  std::function<implicit_signature> implicit_func;
+};
+
+class SecondOrderODE : public mfem::SecondOrderTimeDependentOperator {
+public:
+  using explicit_signature = void(const double, const mfem::Vector&, const mfem::Vector&, mfem::Vector&);
+  using implicit_signature = void(const double, const double, const double, const mfem::Vector&, const mfem::Vector&,
+                                  mfem::Vector&);
+
+  SecondOrderODE(int n) : mfem::SecondOrderTimeDependentOperator(n, 0.0){};
+  void Mult(const mfem::Vector& u, const mfem::Vector& du_dt, mfem::Vector& d2u_dt2) const
+  {
+    explicit_func(t, u, du_dt, d2u_dt2);
+  }
+  void ImplicitSolve(const double dt0, const double dt1, const mfem::Vector& u, const mfem::Vector& du_dt,
+                     mfem::Vector& d2u_dt2)
+  {
+    implicit_func(t, dt0, dt1, u, du_dt, d2u_dt2);
+  }
+
+  std::function<explicit_signature> explicit_func;
+  std::function<implicit_signature> implicit_func;
+};
```

Comment:
Are these used in this PR? Is the main idea to be able to access `Mult` and `ImplicitSolve` via `explicit_func` and `implicit_func`?